src/main/java/com/taskadapter/redmineapi/internal/RedmineJSONParser.java

```diff
@@ -155,6 +155,8 @@ public static TimeEntry parseTimeEntry(JSONObject object)
 		result.setSpentOn(getDateOrNull(object, "spent_on"));
 		result.setCreatedOn(getDateOrNull(object, "created_on"));
 		result.setUpdatedOn(getDateOrNull(object, "updated_on"));
+		final JSONArray custom_fields = JsonInput.getArrayOrNull(object, "custom_fields");
```

Comment:
getArrayOrNull name implies that null will be returned if nothing is found. then later the result of this method is passed to  parseCustomFields()   where it is referenced directly, thus allowing NullPointerException to happen.